[DOCS] add setup guidance for using the Python api with GX Cloud (#9168)
great-expectations/great_expectations: docs/docusaurus/docs/cloud/quickstarts/python_api_quickstart.md

---

> sidebar_label: 'Quickstart for GX Cloud and Python scripts'

Reviewer: ```suggestion
sidebar_label: 'Quickstart for GX Cloud and Python'
```

> title: 'Quickstart for GX Cloud and Python scripts'

Reviewer: ```suggestion
title: 'Quickstart for GX Cloud and Python'
```

id: python_quickstart

> description: Connect to a GX Cloud account from a Python script.

Reviewer: ```suggestion
description: Use Python to set up GX Cloud.
```

Author: The original description is more accurate: They aren't actually setting up GX Cloud through Python -- they are still following the standard setup steps for getting an account; the only difference is that they are setting their environment variables outside of docker.  That still isn't doing any setup in Python, however.  Once they are in Python when they run `get_context()` it will detect those environment variables and initialize a GX Cloud Data Context, which will be connected to their GX Cloud account.

---
import Prerequisites from '/docs/components/_prerequisites.jsx'
import PrereqPython from '/docs/components/prerequisites/_python_version.md'
import SetupAndInstallGx from '/docs/components/setup/link_lists/_setup_and_install_gx.md'
import DataContextInitializeInstantiateSave from '/docs/components/setup/link_lists/_data_context_initialize_instatiate_save.md'

> In this quickstart, you'll learn how to use GX Cloud from a Python script or interpreter, such as a Jupyter Notebook. You'll install Great Expectations, configure your GX Cloud environment variables, connect to sample data, build your first Expectation, validate data, and review the validation results through Python code.

Reviewer: ```suggestion
In this quickstart, you'll learn how to use Python scripts or interpreters, such as a Jupyter Notebook, to set up GX Cloud. You'll install Great Expectations, configure your GX Cloud environment variables, connect to sample data, build your first Expectation, validate data, and review the validation results through Python code.
```

> ## Data validation workflow

Reviewer: ```suggestion
## Data validation workflow
```
Remove the workflow diagram. It’s not necessary or helpful in a quickstart when the right TOC shows the task sequence. Also, this is inconsistent with other topics and adds visual clutter.

Author: Removed in an upcoming commit -- this was a hold over from using the OSS quickstart as a base, which unfortunately is not formatted as tightly as the GX Cloud docs.

The following diagram illustrates the end-to-end GX data validation workflow that you'll implement with this quickstart. Click a workflow step to view the related content.

```mermaid
flowchart LR
%%{init: {"themeVariables": {"fontSize" : "24px"}}}%%

    1(Install\n<b>GX</b>) --> 2("Create a\n<b>Data Context</b>")

    2 --> 3

    3("Connect\nto data</b>")

    3 --> 4

    4("Create a\n<b>Validator</b>") --> 5("Create\n<b>Expectations</b>")

    5 --> 6

    6("Run a\n<b>Checkpoint</b>") --> 7("View\n<b>Validation Results</b>")

%% Link each workflow step to section on page.
click 1 "#install-gx"
click 2 "#create-a-data-context"
click 3 "#connect-to-data"
click 4 "#connect-to-data"
click 5 "#create-expectations"
click 6 "#validate-data"
click 7 "#validate-data"
```

## Prerequisites

> - <PrereqPython />

Reviewer: ```suggestion
- <PrereqPython />
```
Remove this as a prerequisite, as installing Python is included in preparing a user's environment.

Author: Removed in an upcoming commit.

> - pip

Reviewer: ```suggestion
- pip
```
Remove this as a prerequisite, as installing pip is included in preparing a user's environment.

Author: Removed in an upcoming commit.

- An internet browser

> - A GX Cloud Beta Account

Reviewer: ```suggestion
- A GX Cloud Beta Account
```
Link to the GX Cloud Beta [sign-up page](https://greatexpectations.io/cloud), or duplicate the existing procedure and remove this as a prerequisite.

Author: In an upcoming commit: Updated this and the internet access/browser prerequisite to be in line with the existing GX Cloud docs.

    ```bash title="Terminal input"
    pip install great_expectations
    ```

    It can take several minutes for the installation to complete.

## Get your user access token and organization ID

You'll need your user access token and organization ID to set your environment variables. Don't commit your access tokens to your version control software.

1. In GX Cloud, click **Settings** > **Tokens**.

2. In the **User access tokens** pane, click **Create user access token**.

3. Complete the following fields:

    - **Token name** - Enter a name for the token that will help you quickly identify it.

    - **Role** - Select **Admin**. For more information about the available roles, click **Information** (?) .

4. Click **Create**.

5. Copy and then paste the user access token into a temporary file. The token can't be retrieved after you close the dialog.

6. Click **Close**.

7. Copy the value in the **Organization ID** field into the temporary file with your user access token and then save the file. 

    GX recommends deleting the temporary file after you set the environment variables.

## Set the GX Cloud Organization ID and user access token as environment variables

> Environment variables securely store your GX Cloud access credentials. To use GX Cloud in Python scripts: 

Reviewer: ```suggestion
Environment variables securely store your GX Cloud access credentials.
```

1. Save your **GX_CLOUD_ACCESS_TOKEN** and **GX_CLOUD_ORGANIZATION_ID** as environment variables by entering `export ENV_VAR_NAME=env_var_value` in the terminal or adding the command to your `~/.bashrc` or `~/.zshrc` file. For example:

    ```bash title="Terminal input"
    export GX_CLOUD_ACCESS_TOKEN=<user_access_token>
    export GX_CLOUD_ORGANIZATION_ID=<organization_id>
    ```

    :::note

>     Once you have saved your **GX_CLOUD_ACCESS_TOKEN** and **GX_CLOUD_ORGANIZTION_ID** you will be able to access your GX Cloud account in Python scripts and environments while following the [GX OSS guides](../../oss/oss.md).

Reviewer: ```suggestion
   After you save your **GX_CLOUD_ACCESS_TOKEN** and **GX_CLOUD_ORGANIZTION_ID**, you can use Python scripts to access GX Cloud and complete other tasks. See the [GX OSS guides](../../oss/oss.md).
```

    :::

2. Optional. If you created a temporary file to record your user access token and Organization ID, delete it.

## Create a Data Context

- Use the following command in Python to create a Data Context object:

    ```python name="tutorials/quickstart/quickstart.py get_context"
    ```

> ## Connect to data

Reviewer: ```suggestion
## Connect to data
```
For consistency, this should be updated to _Connect to a Data Asset_

> - Use the following command in Python to connect to existing `.csv` data stored in the `great_expectations` GitHub repository and create a Validator object:

Reviewer: ```suggestion
- Run the following Python code to connect to existing `.csv` data stored in the `great_expectations` GitHub repository and create a Validator object:
```
If this edit is accepted, use the same pattern in the other procedure steps.

    ```python name="tutorials/quickstart/quickstart.py connect_to_data"
    ```

    The code example uses the default <TechnicalTag tag="data_context" text="Data Context"/> <TechnicalTag tag="datasource" text="Data Source"/> for Pandas to access the `.csv` data from the file at the specified URL path.

## Create Expectations

> - Use the following commands to create two <TechnicalTag tag="expectation" text="Expectations"/> and save them to the <TechnicalTag tag="expectation_suite" text="Expectation Suite"/>:

Reviewer 2: This is causing the semicolon to be indented t the next time.

    ```python name="tutorials/quickstart/quickstart.py create_expectation"
    ```

  The first Expectation uses domain knowledge (the `pickup_datetime` shouldn't be null).

  The second Expectation uses explicit kwargs along with the `passenger_count` column.

## Validate data

1. Use the following command to define a Checkpoint and examine the data to determine if it matches the defined Expectations:

    ```python name="tutorials/quickstart/quickstart.py create_checkpoint"
    ```

2. Use the following command to return the Validation Results:

    ```python name="tutorials/quickstart/quickstart.py run_checkpoint"
    ```

3. Use the following command to view an HTML representation of the Validation Results in the generated Data Docs:

    ```python name="tutorials/quickstart/quickstart.py view_results"
    ```

## Related documentation

> If you're ready to continue your GX journey, the following topics can help you implement a solution for your specific environment and business requirements:

Reviewer: ```suggestion

```
Don't add unnecessary introductory text. Users understand that the list of topics can be helpful. This is also inconsistent with the existing GX Cloud content.

> - [Install GX in a specific environment with support for a specific Data Source](/docs/oss/guides/setup/installation/install_gx).

Reviewer: ```suggestion
- [Install GX in a specific environment with a specific Data Source](/docs/oss/guides/setup/installation/install_gx)
```

> - [Initialize, instantiate, and save a Data Context](/docs/oss/guides/setup/configure_data_contexts_lp).

Reviewer: ```suggestion
- [Initialize, instantiate, and save a Data Context](/docs/oss/guides/setup/configure_data_contexts_lp)
```

> - [Connect to Data Sources](/docs/oss/guides/connecting_to_your_data/connect_to_data_lp).

Reviewer: ```suggestion
- [Connect to Data Sources](/docs/oss/guides/connecting_to_your_data/connect_to_data_lp)
```

> - [Create and manage Expectations and Expectation Suites](/docs/oss/guides/expectations/expectations_lp/).

Reviewer: ```suggestion
- [Create and manage Expectations and Expectation Suites](/docs/oss/guides/expectations/expectations_lp/)
```

> - [Create, manage, and run Checkpoints](/docs/oss/guides/validation/checkpoints/checkpoint_lp/).

Reviewer: ```suggestion
- [Create, manage, and run Checkpoints](/docs/oss/guides/validation/checkpoints/checkpoint_lp/)
```
